contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -69,6 +69,9 @@ func createReceiverFunction(s *SubscriptionsSupervisor, logger *zap.SugaredLogge
 		ch := getSubject(channel)
 		if err := stanutil.Publish(s.natssConn, ch, &m.Payload, logger); err != nil {
 			logger.Errorf("Error during publish: %v", err)
+			if err.Error() == stan.ErrConnectionClosed.Error() {
+				logger.Fatalf("Connection to NATS was lost, dispatcher is exiting.")
```

Comment:
Add a TODO that states we should reconnect rather than restarting, but for now restarting is better than staying in a broken state forever.

Reply:
Sounds good. Will do.